ambari-server/src/main/resources/common-services/SPARK2/2.0.0/package/scripts/alerts/alert_spark2_thrift_port.py

```diff
@@ -147,22 +147,19 @@ def execute(configurations={}, parameters={}, host_name=None):
         if host_name is None:
             host_name = socket.getfqdn()
 
+        beeline_url = ["jdbc:hive2://{host_name}:{port}/default", "transportMode={transport_mode}"]
         if security_enabled:
+            beeline_url.append("principal={hive_principal}")
+        if transport_mode == "http":
+            beeline_url.append("httpPath=cliservice")
             if spark_ssl_enabled:
-                beeline_url = ['"jdbc:hive2://{host_name}:{port}/default;principal={hive_principal};transportMode={transport_mode};ssl=true;sslTrustStore={spark_truststore_path};trustStorePassword={spark_truststore_pass!p};httpPath=cliservice"']
-            else:
-                beeline_url = ["jdbc:hive2://{host_name}:{port}/default;principal={hive_principal}","transportMode={transport_mode}"]
-        else:
-            if spark_ssl_enabled:
-                beeline_url = ['"jdbc:hive2://{host_name}:{port}/default;transportMode={transport_mode};ssl=true;sslTrustStore={spark_truststore_path};trustStorePassword={spark_truststore_pass!p};httpPath=cliservice"']
-            else:
-                beeline_url = ["jdbc:hive2://{host_name}:{port}/default","transportMode={transport_mode}"]
-                
+                beeline_url.append("ssl=true;sslTrustStore={spark_truststore_path};trustStorePassword={spark_truststore_pass!p}")
```

Comment:
this is not critical, just a remark: as now we building the url from array, it would be great to do it one fashion. For me it would be more clean to use here `beeline_url.extend([....])`. Also it would not hard-code the delimiter in several places.